chef-config/lib/chef-config/config.rb

```diff
@@ -155,6 +162,22 @@ def self.is_valid_url?(uri)
     # properly.
     configurable(:daemonize).writes_value { |v| v }
 
+    def self.expand_relative_paths(path)
+      unless path.nil?
+        if path.is_a?(String)
+          File.expand_path(path)
+        else
+          path.map do |path|
+            File.expand_path(path)
+          end
+        end
```

Comment:
You should be able to just `Array(path).map { |path| File.expand_path(path) }`

Reply:
that actually replaces the whole block:

```
unless path.nil?
  Array(path).map { |path| File.expand_path(path) }
end
```

if its a string then `Array(path)` wraps an Array around it, but leaves any Array that it finds, and means that there's no `.is_a?` check anymore which means that any scalar type which File.expand_path operates on will work with `expand_relative_paths` without needing to add more types to the check, and is clearer than inverting that and doing an `.is_a?` check on the Array instead.  the use of `is_a?` is also always brittle and creates tight coupling.

Reply:
actually this works:

```ruby
def self.expand_relative_paths(path)
  Array(path).map { |path| File.expand_path(path) }
end
```

because:

```
# pry
[1] pry(main)> Array(nil)
=> []
```

Reply:
@lamont-granquist If we have string value for any option eg: file_cache_path = "~/test/.chef/cache"
then it will set file_cache_path = ["/root/test/.chef/cache"]
So this might result in the bug and potentially cause other issues as well. 

If there are any more suggestions considering my comment please share.

Reply:
I see.  Yeah, that might break downstream consumers external to our code that are buggy against arrays of dirs here (which would still be bugs on their side, but we'd break them hard if we did this).

Okay.